geode-core/src/main/java/org/apache/geode/internal/cache/BucketAdvisor.java

```diff
@@ -1605,21 +1605,34 @@ protected Profile instantiateProfile(InternalDistributedMember memberId, int ver
       InternalCache cache = getProxyBucketRegion().getCache();
       List<CacheServer> servers = cache.getCacheServers();
```

Comment:
The "cache" and "servers" are only used in the new getBucketServerLocations so move them into that method

Reply:
Done